type-class-schema/src/main/scala/ru/primetalk/typed/ontology/typeclass/schema/Replace.scala

```diff
@@ -0,0 +1,14 @@
+package ru.primetalk.typed.ontology.typeclass.schema
+
+type Replace[S <: Tuple, FROM, TO] <: Tuple = S match {
+  case EmptyTuple => EmptyTuple
+  case FROM *: t => TO *: t
+  case h *: t => h *: Replace[t, FROM, TO]
+}
+
+inline def replace[S <: Tuple, FROM <: Singleton, TO <: Singleton](inline s: S, inline from: FROM, inline to: TO): Replace[S, FROM, TO] =
```

Comment:
a.troitskiy recommends to try using erasedValue: https://scastie.scala-lang.org/ywmmO1E5QcaZdu6BosiJZQ